app/src/main/java/com/github/gotify/login/LoginActivity.java

```diff
@@ -100,6 +105,7 @@ private void invalidateUrl() {
         checkUrlButton.setText(getString(R.string.check_url));
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.O)
```

Comment:
This method must not have the RequiredApi annotation, wrap the client cert code into an if similar to

https://github.com/gotify/android/blob/7fd8ee06e1d3ccbc1d94cc59594a0f64dc887c82/app/src/main/java/com/github/gotify/init/InitializationActivity.java#L45-L48

The client cert options inside the advanced settings dialog should be hidden / disabled if `Build.VERSION.SDK_INT < Build.VERSION_CODES.O`

Currently, if an android version less than `Build.VERSION_CODES.O` is used, then the app will crash when base64 stuff is accessed, because these classes don't exist in older android versions.

Reply:
Ok, those guards have been put into place. Tested on 4a API 30 and 3 API 24 and everything works against my server with and without client certificates respectively.